clang/lib/AST/ExprCXX.cpp

```diff
@@ -1013,15 +1013,15 @@ CXXDefaultArgExpr *CXXDefaultArgExpr::CreateEmpty(const ASTContext &C,
   return new (Mem) CXXDefaultArgExpr(EmptyShell(), HasRewrittenInit);
 }
 
-CXXDefaultArgExpr *CXXDefaultArgExpr::Create(const ASTContext &C,
-                                             SourceLocation Loc,
-                                             ParmVarDecl *Param,
-                                             Expr *RewrittenExpr,
-                                             DeclContext *UsedContext) {
+CXXDefaultArgExpr *
+CXXDefaultArgExpr::Create(const ASTContext &C, SourceLocation Loc,
+                          ParmVarDecl *Param, DeclContext *UsedContext,
+                          Expr *RewrittenExpr, bool HasRewrittenInit) {
   size_t Size = totalSizeToAlloc<Expr *>(RewrittenExpr != nullptr);
```

Comment:
Maybe the naming here is just weird, but is there ever a case where `RewrittenExpr != nullptr` and `HasRewrittenInit` *don’t* evaluated to the same value? Because in that case, either `HasRewrittenInit` seems superfluous or `RewrittenExpr` is badly named, because a ‘non-rewritten `RewrittenExpr`’ sounds a bit nonsensical...

Reply:
I have the same question!
In particular in the test change, I think the expression _is_ rewritten.